### ISSUE TITLE
Version 23.97.0 Authentication failure with gitlab (23.96.3 working)

### ISSUE THREAD

**Issue by mfechner** [renovatebot/renovate#7923]
**What Renovate type, platform and version are you using?**

Renovate is running on a self-hosted gitlab installation.

**Describe the bug**

Till version 23.96.3 everything was fine. Starting with 23.97.0 I get a:
```
Executing "step_script" stage of the job script
03:54
$ npm run renovate
> gitlab-renovate-pipeline@1.0.0 renovate /var/tmp/gitlab_runner/builds/sz8rQmBx/0/mfechner/renovate
> renovate --platform gitlab --endpoint https://gitlab.fechner.net/api/v4 $(cat repositories.txt | xargs)
FATAL: Authentication failure
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "logContext": "xPbiJ9WHi",
           "msg": "Authentication failure"
         }
       ]
npm ERR! code ELIFECYCLE
npm ERR! errno 1
```

**Additional context**

I checked the changelog, but there is nothing mentioned that something related to gitlab auth was changed:
```
Features

gradle-lite: JS-based Gradle manager (#​7521) (c43a4dc)

Code Refactoring
build: Pre-compile files imported from data folder (#​7896) (31c7d8a)
```

Was there something changed that is not documented?

**Comment by HonkingGoose** [comment]
@rarkins, I think this user is maybe running into #7843? Maybe the user should migrate to the `renovate-runner` pipeline bot on GitLab?

[Renovate docs on migrating to self-hosted pipeline on GitLab](https://docs.renovatebot.com/gitlab-bot-security/#recommended-migration)
[renovate-runner project on GitLab](https://gitlab.com/renovate-bot/renovate-runner)

**Comment by viceice** [comment]
@HonkingGoose  No, I think this is a different issue, as he already uses a self-hosted renovate

@mfechner Can you verify the older renovate version is still working?

**Comment by rarkins** [comment]
The changelog from 23.96.3 to 23.97.0 is here: https://github.com/renovatebot/renovate/compare/23.96.3...23.97.0

It is highly unlikely to have caused an authentication problem. More likely is that the token you're using got revoked or expired, or something else altogether.

**Comment by viceice** [comment]
@mfechner Please rerun with `LOG_LEVEL=debug npm run renovate`, so we can see the early debug logs.

**Comment by mfechner** [comment]
@rarkins I think you are right.
I think the error message `FATAL: Authentication failure` is really misleading.

I saw that my public ID got blocked by the firewall due to many failed logins. As renovate is using the public IP to access the public IP, the connection from renovate to `https://gitlab.fechner.net` got blocked too.

I unblocked the IP again and now renovate is running again. The blocking of the IP match unluckily the time of the upgrade.

Maybe the error message should be more clear here?

**Comment by mfechner** [comment]
Maybe it should report here `Connection failed` and not `FATAL: Authentication failure`?
